core/provider/definitions.go

```diff
@@ -8,6 +13,22 @@ type VolumeDefinition struct {
 	Size      string
 }
 
+func (v VolumeDefinition) ValidateBasic() error {
```

Comment:
```suggestion
func (v *VolumeDefinition) ValidateBasic() error {
```

These can all be pointer receiver methods.